froide_payment/models.py

```diff
@@ -188,8 +188,20 @@ def create_recurring_order(self, force=False, now=None, remote_reference=None):
         )
 
     def create_order(
-        self, kind="", description=None, is_donation=True, remote_reference=""
+        self,
+        kind="",
+        description=None,
+        is_donation=True,
+        remote_reference="",
+        remote_reference_is_unique=False,
     ):
+        existing_order = Order.objects.filter(remote_reference=remote_reference)
+        if remote_reference_is_unique and existing_order.exists():
+            raise ValueError(
```

Comment:
I'm wondering if this should then just return that order?

Reply:
I wasn't sure, but decided against it to keep with the semantics of an `create_` function and not change this into a `get_or_create`. This also gives us the option to find the errors laters and fix the idempotency problems that lead to this errorneous behaviour